import-export-cli/integration/testutils/environment_testUtils.go

```diff
@@ -411,3 +411,234 @@ func validateMutualSSLCerts(t *testing.T, apiParams *Params, path string) {
+			singleProductionEndpointInApi := productionEndpointsInApi[i].(map[string]interface{})
+			assert.Equal(t, productionEndpointsInParams[i]["url"], singleProductionEndpointInApi["url"])
+			assert.Equal(t, "address", singleProductionEndpointInApi["endpoint_type"])
+		}
+	}
+
+	// Validate Sandbox endpoints
+	sandboxEndpointsInParams := apiParams.Environments[0].Configs.LoadBalanceEndpoints.Sandbox
+	sandboxEndpointsInApi := endPointConfigInApi.(map[string]interface{})["sandbox_endpoints"].([]interface{})
+	if strings.EqualFold(apiParams.Environments[0].Configs.LoadBalanceEndpoints.SessionManagement, "transport") {
+		for i := 0; i < len(sandboxEndpointsInParams); i++ {
+			singleSandboxEndpointInApi := sandboxEndpointsInApi[i].(map[string]interface{})
+			assert.Equal(t, sandboxEndpointsInParams[i]["url"], singleSandboxEndpointInApi["url"])
+		}
+	}
+	if strings.EqualFold(apiParams.Environments[0].Configs.LoadBalanceEndpoints.SessionManagement, "soap") {
+		for i := 0; i < len(sandboxEndpointsInParams); i++ {
+			singleSandboxEndpointInApi := sandboxEndpointsInApi[i].(map[string]interface{})
+			assert.Equal(t, sandboxEndpointsInParams[i]["url"], singleSandboxEndpointInApi["url"])
+		}
+	}
+
+	same := "override_with_same_value"
+	api.SetEndpointType(same)
+	importedAPI.SetEndpointType(same)
+	api.SetEndPointConfig(same)
+	importedAPI.SetEndPointConfig(same)
+
+	ValidateAPIsEqual(t, api, importedAPI)
+}
+
+func ValidateHttpEndpointWithFailover(t *testing.T, api *apim.API, apiParams *Params, importedAPI *apim.API) {
+	t.Helper()
+
+	var isSaopEndpoint bool = false
+
+	//Validate EndPoint Type
+	assert.Equal(t, "failover", importedAPI.GetEndpointType())
+
+	endPointConfigInApi := importedAPI.EndpointConfig
+	//Check whether the endpoint are SOAP endpoints
+	if strings.EqualFold(apiParams.Environments[0].Configs.EndpointType, "soap") {
+		isSaopEndpoint = true
+	}
+
+	// Validate Production endpoints
+	productionEndpointInParams := apiParams.Environments[0].Configs.FailoverEndpoints.Production
+	productionEndpointInApi := endPointConfigInApi.(map[string]interface{})["production_endpoints"].(map[string]interface{})
+	assert.Equal(t, productionEndpointInParams.URL, productionEndpointInApi["url"])
+	if isSaopEndpoint {
+		assert.Equal(t, "address", productionEndpointInApi["endpoint_type"])
+	}
+
+	// Validate Production failover endpoints
+	productionFailoverEndpointsInParams := apiParams.Environments[0].Configs.FailoverEndpoints.ProductionFailovers
+	productionFailoverEndpointsInApi := endPointConfigInApi.(map[string]interface{})["production_failovers"].([]interface{})
+	for i := 0; i < len(productionFailoverEndpointsInParams); i++ {
+		singleProductionFailoverEndpointInApi := productionFailoverEndpointsInApi[i].(map[string]interface{})
+		assert.Equal(t, productionFailoverEndpointsInParams[i]["url"], singleProductionFailoverEndpointInApi["url"])
+		if isSaopEndpoint {
+			assert.Equal(t, "address", productionEndpointInApi["endpoint_type"])
+		}
+	}
```

Comment:
This code segment is also similar to the code in 561 to 568. Shall we use a separate function here as well and reuse it in both the places to reduce the number of lines?